xilem_core/examples/filesystem.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2024 the Xilem Authors
```

Comment:
This is incomplete - need to add folders

Help welcome.

This is a bit messy, because it's not completely clear what the source of truth with the filesystem is. (We will likely have a similar issue with DOM nodes)